discordia/containers/Embed.lua

```diff
@@ -0,0 +1,35 @@
+local Snowflake = require('./Snowflake')
+
+local format = string.format
+
+local Embed, property, method = class('Embed', Snowflake)
+Embed.__description = "Represents Discord Embed."
+
+function Embed:__init(data, parent)
+	Snowflake.__init(self, data, parent)
+	self._enabled = data.enabled
+	if not self._enabled then return end 
+	self._channel_id = data.channel_id
+end
+
+function Embed:__tostring()
+	return format('%s: %s', self.__name, self._channel_id or "") -- will not always have a channel id
+end
+
+
+local function getGuild(self)
+	return self._parent._api:getGuildEmbed(self._guild_id)
+end
+
+local function modifyGuild(self, enabled, channel_id)
```

Comment:
It is called "modifyGuild" because what you are modifying is that guild's Embed. Said that, I understand what you mean by splitting into different methods and I will add it right now.